yt_dlp/extractor/playsuisse.py

```diff
@@ -135,11 +138,29 @@ class PlaySuisseIE(InfoExtractor):
             url
         }'''
 
+    def _perform_login(self, username, password):
+        if username is None or password is None:
+            self.raise_login_required('Your Play Suisse account\'s details must be provided with --username and --password.')
+
+        login_page = self._download_webpage('https://www.playsuisse.ch/api/sso/login?x=x&locale=de&redirectUrl=https%3A%2F%2Fwww.playsuisse.ch%2F', None, note="Downloading login page")
+        settings_str = re.search("^var SETTINGS(.*)", login_page, flags=re.MULTILINE).group()[len("var SETTINGS = "):-2]
+        settings = json.loads(settings_str)
+
+        login_endpoint = "https://login.srgssr.ch/srgssrlogin.onmicrosoft.com/B2C_1A__SignInV2/SelfAsserted?tx=" + settings['transId'] + "&p=B2C_1A__SignInV2"
+        webpage, urlh = self._download_webpage_handle(login_endpoint, None,
+                                                      note="Logging in",
+                                                      headers={"X-CSRF-TOKEN": settings['csrf']},
+                                                      data=urlencode_postdata({"request_type": "RESPONSE", "signInName": username, "password": password}))
+
+        loginconfirmed_url = "https://login.srgssr.ch/srgssrlogin.onmicrosoft.com/B2C_1A__SignInV2/api/CombinedSigninAndSignup/confirmed?rememberMe=false&csrf_token=" + settings['csrf'] + "&tx=" + settings['transId'] + "&p=B2C_1A__SignInV2&diags="
+        webpage, urlh = self._download_webpage_handle(loginconfirmed_url, None, note="Downloading ID token")
+        self._id_token = parse_qs(urlsplit(urlh.url).query)['id_token'][0]
```

Comment:
use `yt_dlp.utils.parse_qs` instead, then you won't need to `urlsplit`

we should probably error gracefully here (`raise ExtractorError`) if an `id_token` value is not present in the query instead of KeyErroring